tools/mo/openvino/tools/mo/front/kaldi/add_reshape_transpose_around_conv_pool.py

```diff
@@ -0,0 +1,219 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+import networkx as nx
+
+from openvino.tools.mo.front.common.partial_infer.utils import int64_array
+from openvino.tools.mo.front.common.replacement import FrontReplacementPattern
+from openvino.tools.mo.front.tf.graph_utils import create_op_with_const_inputs, create_op_node_with_second_input
+from openvino.tools.mo.graph.graph import Graph, Node, Port
+from openvino.tools.mo.ops.concat import Concat
+from openvino.tools.mo.ops.elementwise import Div
+from openvino.tools.mo.ops.reshape import Reshape
+from openvino.tools.mo.ops.shape import Shape
+from openvino.tools.mo.ops.transpose import Transpose
+from openvino.tools.mo.utils.shape import node_to_get_shape_value_of_indices
+
+
+def find_max_frame_time(node: Node):
+    """
+    Find maximum time_dim among all inputs of given node
+    If one of inputs has undefined time_dim, it returns None.
+    """
+    in_frame_time_max = 0
+
+    for inp in node.in_ports():
+        if node.in_port(inp).disconnected():
+            continue
+        in_node = node.in_port(inp).get_source().node
+        if in_node.time_dim == -1:
+            return None
+        if in_node.time_dim > in_frame_time_max:
+            in_frame_time_max = in_node.time_dim
+
+    return in_frame_time_max
+
+
+def propagate_time_dim_through_branch(child_node: Node, dest_port: Port):
```

Comment:
I meant to use `child_node` instead of `dest_port`. So you will have:
```
def propagate_time_dim_through_branch(parent_node: Node, child_node: Node):
```